synapse/handlers/__init__.py

```diff
@@ -54,7 +55,12 @@ def __init__(self, hs):
         self.directory_handler = DirectoryHandler(hs)
         self.typing_notification_handler = TypingNotificationHandler(hs)
         self.admin_handler = AdminHandler(hs)
+        asapi = ApplicationServiceApi(hs)
         self.appservice_handler = ApplicationServicesHandler(
-            hs, ApplicationServiceApi(hs)
+            hs, asapi, AppServiceScheduler(
+                clock=hs.get_clock(),
+                store=hs.get_datastore(),
+                as_api=asapi
+            )
```

Comment:
Yup, for testing. I don't think putting it in the ``HomeServer`` object makes it that much clearer, given the code which runs this is:
```
def build_handlers(self):
    return Handlers(self)
```
I'd rather keep the number of hops as low as possible without adding the ``HomeServer`` implementation into the DI chain.